### DISCUSSION TITLE
Novas Web APIs | Payment Request

### DISCUSSION THREAD

**Discussion by grvcoelho** [frontendbr/forum#340] [Dicas]
Galera,

Eu sempre tento ficar atualizado com as novas Web APIs e semana passada eu fiz uns experimentos com uma bem interessante: a **Payment Request API**.

Basicamente, é uma API que facilita o checkout de pagamentos para os sites com uma UI nativa e bonitona dos browsers.

Eu gravei um vídeo no Pagar.me Talks, explicando em detalhes como esta API funciona: https://www.youtube.com/watch?v=kD0j2BwgF7c

[![](https://cloud.githubusercontent.com/assets/7416751/19725476/63c78382-9b64-11e6-92ad-20df48e8300e.png)](https://www.youtube.com/watch?v=kD0j2BwgF7c)

Eu também fiz uma demo (só clicar na imagem abaixo, no chrome +53 do mobile) acessem no chrome +53 do Android). E o código está aqui no github: https://github.com/grvcoelho/payment-request

[![](https://cloud.githubusercontent.com/assets/7416751/19725552/da94fb34-9b64-11e6-8f6c-f8474d761dfa.png)](https://grvcoelho.github.io/payment-request)

---

Para saber mais:
- Aqui está a especificação da Payment Request API da W3C: https://w3c.github.io/browser-payment-api/
- Aqui tem um guia de integração do Google Web Fundamentals: https://developers.google.com/web/fundamentals/getting-started/primers/payment-request/

Aliás, se vocês ainda não leram, super recomendo ler toda documentação do Google Web Fundamentals (https://developers.google.com/web/fundamentals/)

---

Busquem conhecimento :smile:

Abs

**Comment by robsonbittencourt** [comment]
Não assisti o video ainda, mas já deixo os parabéns pois venho acompanhando os videos que vocês tem feito ai na Pagar.me.

Mas já adianto uma dúvida @grvcoelho. Como fica a segurança? Estes dados ficam armazenados no dispositivo pelo que pude perceber na sua demo. O que impede de alguém roubá-los?

Falando nisso fiquei com medo da demo hahaha. Já identificou meu cartão de crédito :)

**Comment by grvcoelho** [comment]
@LFeh: adiciona também as tags `HTML` e `Javascript`, please? :smile:

**Comment by grvcoelho** [comment]
Oi @robsonbittencourt, tudo bem? Que bom que você tá curtindo os vídeos:

Tem dois pontos aí na sua mensagem:

1) sobre segurança: os dados do cartão ficam criptografados e só são passados para quem está usando a api se o `cvv` (que é o código de segurança do cartão) estiver correto. Se passar algum outro cvv, a request vai falhar :smile:

2) sobre a demo identificar teu cartão já: pode ter dois motivos.
a) Você já usou algum outro site com payment request api
b) O chrome tem um negócio que já está rodando há algum tempo que te pergunta, em checkouts normais de lojas mesmo, se você quer salvar o cartão no browser. Depois ele usa isso para fazer autocomplete de inputs. E ele usa isso também no Payment Request API :D

**Comment by felipefialho** [comment]
Que aula hein? Show!
